cmake/UnitTests.cmake

```diff
@@ -391,6 +391,13 @@ list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipMallocPitch_KernelLaunch - float")
 list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipMallocPitch_KernelLaunch - double") # Failed
 list(APPEND CPU_OPENCL_FAILED_TESTS "Unit_hipGraphAddHostNode_ClonedGraphwithHostNode") # Failed
 list(APPEND CPU_OPENCL_FAILED_TESTS "syncthreadsExitedThreads") # Timeout
+# The following warning appears in the CI with LLVM 16:
+#
+#   warning: stack frame size (18446744073709551608) exceeds limit (4294967295) in function '_Z15matrixTransposePfS_i'
+#
+# altought the kernels in the SPIR-V code do not have stack objects at
+# all and, otherwise, there does not seem to be anything out of ordinary.
+list(APPEND CPU_OPENCL_FAILED_TESTS "dynamic_shared" "hipDynamicShared" "cuda-template") # SEGFAULT
```

Comment:
Where is this segfault coming from if "there does not seem to be anything out of ordinary."?

Reply:
No idea. All I know is that the segfault appears in the CI along with the warning message.

Reply:
I'm not sure if just adding more and more issues into the excluded test list is a good idea. Main already seems to be broken I think we should resolve issues in main or at the very least not introduce any more bugs.